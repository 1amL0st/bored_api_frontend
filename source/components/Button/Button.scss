.button {
  background-color: lightgray;
  color: black;
  padding: 0.7em;

  min-width: 128px;
  min-height: 32px;

  border-radius: 16px;
  outline: none;
  border: none;

  font-weight: bold;
  text-transform: uppercase;

  &:hover {
    cursor: pointer;
  }
}