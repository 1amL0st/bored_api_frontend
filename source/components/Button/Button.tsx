import React from 'react';
import classNames from 'classnames';

import './Button.scss';

interface IProps {
  className?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  children: React.ReactNode;
}

export const Button = ({ className, ...props }: IProps) => {
  return <button className={classNames('button', className)} {...props} />;
};
