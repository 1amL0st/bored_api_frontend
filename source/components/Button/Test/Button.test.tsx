import React from 'react';
import * as renderer from 'react-test-renderer';
import { Button } from '../Button';

describe('Components: Button test', () => {
  test('First test', () => {
    const component = renderer.create(<Button>Facebook</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Second test', () => {
    const component = renderer.create(
      <Button>
        <p>Some text here</p>
      </Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
