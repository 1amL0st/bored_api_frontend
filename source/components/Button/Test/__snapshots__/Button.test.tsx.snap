// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Button test First test 1`] = `
<button
  className="button"
>
  Facebook
</button>
`;

exports[`Components: Button test Second test 1`] = `
<button
  className="button"
>
  <p>
    Some text here
  </p>
</button>
`;
